### PR TITLE
tempdir: Add a `persist` method

### PR DESCRIPTION
**Pull request by cgwalters**
I have a use case where I want the equivalent of `into_path()`.

### CODE REVIEW

File: cap-tempfile/src/lib.rs [hunk 1]
```diff
@@ -101,6 +101,15 @@ impl TempDir {
...
     }
 
+    /// Make this directory persistent.
+    ///
+    /// This corresponds to [`tempfile::TempDir::into_path`].
+    ///
+    /// [`tempfile::TempDir::into_path`]: https://docs.rs/tempfile/latest/tempfile/struct.TempDir.html#method.into_path
+    pub fn persist(mut self) -> io::Result<Dir> {
+        Ok(self.dir.take().unwrap())
+    }
```
Comment (sunfishcode) [inline cap-tempfile/src/lib.rs:111]: Would it make sense to call this function `into_dir`, for closer analogy with the tempfile API?

File: cap-tempfile/src/lib.rs [hunk 2]
```diff
@@ -101,6 +101,15 @@ impl TempDir {
         Err(Self::already_exists())
     }
 
+    /// Make this directory persistent.
+    ///
+    /// This corresponds to [`tempfile::TempDir::into_path`].
```
Comment (sunfishcode) [inline cap-tempfile/src/lib.rs:106]: Could you add ", except that it returns a `Dir` instead of a `PathBuf`" to this comment?

**Review by sunfishcode — COMMENTED** [review]
Overall looks good, just a few minor comments.

File: cap-tempfile/src/lib.rs (hunk 1 above)
Comment (cgwalters) [reply cap-tempfile/src/lib.rs:112]: Sure, done!

**Comment by sunfishcode** [comment]
Thanks!

**Comment by sunfishcode** [comment]
This is now released in cap-std 3.3.0.

**Pull request merged**